Source/Dafny/DafnyOptions.cs

```diff
@@ -588,41 +584,82 @@ public enum IncludesModes { None, Immediate, Transitive }
       TODO
 
     {:trigger}
-      TODO
-");
-    }
-
+      TODO";
 
     /// <summary>
-    /// Dafny comes with it's own copy of z3, to save new users the trouble of having to install extra dependency.
-    /// For this to work, Dafny makes the Z3ExecutablePath point to the path were Z3 is put by our release script.
-    /// For developers though (and people getting this from source), it's convenient to be able to run right away,
-    /// so we vendor a Windows version.  This is the default value; it may be overwritten by command-line arguments
+    /// Dafny releases come with their own copy of Z3, to save users the trouble of having to install extra dependencies.
+    /// For this to work, Dafny looks for Z3 at the location where it is put by our release script (i.e., z3/bin/z3[.exe]).
+    /// If Z3 is not found there, Dafny relies on Boogie to locate Z3 (which also supports setting a path explicitly on the command line).
+    /// Developers (and people getting Dafny from source) need to install an appropriate version of Z3 themselves.
     /// </summary>
-    public void SetZ3ExecutableName() {
-      var platform = (int)System.Environment.OSVersion.Platform;
+    private void SetZ3ExecutablePath() {
+      var pp = "PROVER_PATH=";
+      var proverPathOption = ProverOptions.Find(o => o.StartsWith(pp));
+      if (proverPathOption != null) {
+        var proverPath = proverPathOption.Substring(pp.Length);
+        // Boogie will perform the ultimate test to see if "proverPath" is real--it will attempt to run it.
+        // However, by at least checking if the file exists, we can produce a better error message in common scenarios.
+        if (!File.Exists(proverPath)) {
+          throw new Bpl.ProverException($"Requested prover not found: '{proverPath}'");
+        }
+      } else {
+        var platform = (int)System.Environment.OSVersion.Platform;
 
-      // http://www.mono-project.com/docs/faq/technical/
-      var isUnix = platform == 4 || platform == 6 || platform == 128;
+        // http://www.mono-project.com/docs/faq/technical/
```

Comment:
Consider removing this outdated comment now we're here.

Reply:
Good idea. Will do in my next (unrelated) PR.